### DISCUSSION TITLE
Is there a way to access mix theme data outside mix styles ?

### DISCUSSION THREAD

**Discussion by jean-smaug** [btwld/mix#225] [Q&A] [answered]
I have a color defined the same way of [this doc section](https://www.fluttermix.com/docs/guides/theming#creating-your-own-design-tokens)

```dart
const primaryColor = ColorToken('primary');

final theme = MixThemeData(
  colors: {
    primaryColor: Colors.blue,
  },
)
```

I would like to access `primaryColor` in a Flutter widget that isn't using the Mix style system ?
Is it possible ?

**Comment by jean-smaug** [comment]
I have something, if it can help someone 🤷‍♂️ 

```dart

  @override
  Widget build(BuildContext context) {
    var primaryColor = MixTokenResolver(context).colorToken(primaryColor)

    // ...
  }
```

Is there any risks in terms of performances doing it ?

**Reply by leoafarias** [reply, marked as answer]
@jean-smaug This is actually something we have been discussing internally. How to resolve styles and/or attributes and not just tokens. I believe this works fine, we will have a bit cleaner implementation that includes attributes also.

**Reply by jean-smaug** [reply]
Super thanks !
BTW wanted to thank you for your great lib !
I'm new to Flutter ecosystem but it deserve to gain in popularity, styling is super convenient 👏 

I read about the custom widget and I find it a little bit complicated to setup 😅 
Do you plan an official support of `TextField` or any kind of inputs ? 
Maybe hard to figure out with libraries like [flutter_form_builder](https://pub.dev/packages/flutter_form_builder) that have custom implem.

I also found the [remix_ui](https://github.com/conceptadev/remix_ui) repo, will it be a complete design system that dev can use or it's for testing purpose ?

**Reply by leoafarias** [reply]
@jean-smaug we will be sharing more about remix_ui soon, but the goal is to be headless widgets to build custom design systems
